aclk/aclk_otp.c

```diff
@@ -444,18 +444,24 @@ static int private_decrypt(RSA *p_key, unsigned char * enc_data, int data_len, u
     if (!ctx)
         return 1;
 
-    if (EVP_PKEY_decrypt_init(ctx) <= 0)
+    if (EVP_PKEY_decrypt_init(ctx) <= 0) {
+        EVP_PKEY_CTX_free(ctx);
         return 1;
+    }
 
-    if (EVP_PKEY_CTX_set_rsa_padding(ctx, RSA_PKCS1_OAEP_PADDING) <= 0)
+    if (EVP_PKEY_CTX_set_rsa_padding(ctx, RSA_PKCS1_OAEP_PADDING) <= 0) {
+        EVP_PKEY_CTX_free(ctx);
         return 1;
+    }
 
     *decrypted = mallocz(outlen);
```

Comment:
valgrind did not report nothing for contexts we are cleaning, but it still reporting for this line:

```sh
==15607== 24 bytes in 1 blocks are still reachable in loss record 1,938 of 5,613
==15607==    at 0x48427C4: malloc (vg_replace_malloc.c:381)
==15607==    by 0x4C92C68: CRYPTO_zalloc (in /lib64/libcrypto.so.3)
==15607==    by 0x4B81ECE: BN_new (in /lib64/libcrypto.so.3)
==15607==    by 0x4B7B3EC: BN_BLINDING_create_param (in /lib64/libcrypto.so.3)
==15607==    by 0x4D87806: RSA_setup_blinding (in /lib64/libcrypto.so.3)
==15607==    by 0x4D8DD82: ??? (in /lib64/libcrypto.so.3)
==15607==    by 0x4DFB16C: ??? (in /lib64/libcrypto.so.3)
==15607==    by 0x5DD323: private_decrypt (aclk_otp.c:459)
==15607==    by 0x5DE68E: aclk_get_mqtt_otp (aclk_otp.c:502)
==15607==    by 0x5D6513: aclk_attempt_to_connect (aclk.c:584)
==15607==    by 0x5D71ED: aclk_main (aclk.c:735)
==15607==    by 0x462DEE: netdata_thread_init (threads.c:262)
```

I am going to approve this PR, but we need to take a look on this line later.